homeassistant/components/zwave/node_entity.py

```diff
@@ -123,6 +125,13 @@ def get_node_statistics(self):
         return self._network.manager.getNodeStatistics(
             self._network.home_id, self.node_id)
 
+    def get_current_state(self):
+        """Retrieve current state of a binary switch"""
+        if self.node.has_command_class(COMMAND_CLASS_SWITCH_BINARY):
+            for value in self.node.get_values().values():
```

Comment:
What's the point of this?

A node can have several switches and you are pulling the data of the first value, not even necessarily COMMAND_CLASS_SWITCH_BINARY

Reply:
@andrey-git all i'm trying to do is get the direction of the switch that was pressed (true/false).  Is there a better way to do that?

Reply:
We should just send an event when a value changes similar to the node event. There shouldn't be any binary switch command class logic here.

https://github.com/home-assistant/home-assistant/blob/96fb7fe84cb5db65398f7edd893b7602cafff7b0/homeassistant/components/zwave/node_entity.py#L166-L175